app/analyzers/terms.py

```diff
@@ -0,0 +1,256 @@
+import numpy as np
+import random
+
+from helpers.singletons import settings, es, logging
+from collections import defaultdict
+from collections import Counter
+import helpers.utils
+from helpers.analyzer import Analyzer
+
+
+class TermsAnalyzer(Analyzer):
+
+    def evaluate_model(self):
```

Comment:
This method is pretty huge, is there a possiblity to split up some of the brute forcing stuff to cut it down in size a bit?